File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.cc
#include<stdio.h>
#define MAX 1000001

int a[MAX];

int main(void){
	a[1]=1;
	for(int i=2; i<MAX; i++){
		for(int j= i+i; j<MAX; j+=i){
			if(a[j]==1) continue;
			else a[j]=1;
		}
	}
	
	while(1){
	int n;
	int count=0;
	scanf("%d" ,&n);
	if(n==0) break;
	for(int i=2; i<n; i++){
		
			if(a[i]==0 && a[n-i]==0 ){
				count++;
				if(count==1){
		        printf("%d = %d + %d\n", n, i, n-i);
		        break;
		    }
			}
		 
}
		if(count ==0) printf("Goldbach's conjecture is wrong.");
			
}
return 0;
}

File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');
const numbers = input.map(Number).filter(num => num > 0); // 0 제거
const maxNum = Math.max(...numbers);

const mapping = Array.from({ length: maxNum + 1 }, () => true);
mapping[0] = false;
mapping[1] = false;

for (let i = 2; i * i <= maxNum; i++) {
    if (mapping[i]) {
        for (let j = i * i; j <= maxNum; j += i) {
            mapping[j] = false;
        }
    }
}

let ans = "";

for (const number of numbers) {
    let found = false;
    for (let i = 3; i <= number; i++) {
        if (mapping[i] && mapping[number - i]) {
            ans += `${number} = ${i} + ${number - i}\n`;
            found = true;
            break;
        }
    }

    if (!found) {
        ans += "Goldbach's conjecture is wrong.\n";
    }
}

console.log(ans);
